tests/test_database.py

```diff
@@ -0,0 +1,68 @@
+import unittest
+from os import remove
+from os.path import isfile
+
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+
+from space_allocator.app import dojo, database, Staff, Fellow, LivingSpace, Office, Base, RoomDB, PersonDB
+
+
+class TestSaveState(unittest.TestCase):
+
+	def setUp(self):
+		self.dojo = dojo
+		self.database = database
+		
+	def test_save_state_file_exists_defined_db(self):
+		"""test that the db is created with specified name"""
+		self.database.save_state("trial.db")
+		self.assertTrue(isfile("trial.db"))
+		remove("trial.db")
+
+	def test_save_state_succesfully_with_defined_db_2(self):
+		"""test that the db is created with specified name"""
+		self.dojo.load_people("dummy.txt")
+		self.database.save_state("trial.db")
+		engine = create_engine("sqlite:///trial.db", echo = False)
+		Base.metadata.bind = engine
+		DBSession = sessionmaker(bind=engine)
+		session = DBSession()
+		dbpersons = session.query(PersonDB).all()
+		self.assertEqual(len(dbpersons), 35)
+		self.assertEqual(dbpersons[17].name, "Skip Cheru")
+
+	def test_save_state_succesfully_with_undefined_db(self):
+		self.dojo.load_people("dummy.txt")
+		self.database.save_state()
+		engine = create_engine("sqlite:///dojo.db", echo = False)
+		Base.metadata.bind = engine
+		DBSession = sessionmaker(bind=engine)
+		session = DBSession()
+		dbpersons = session.query(PersonDB).all()
+		self.assertEqual(len(dbpersons), 35)
+		self.assertEqual(dbpersons[17].name, "Skip Cheru")
+
```

Comment:
Good job querying the database, just fix the small issue in setup, i think the rest you good. I have learned from this